src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenidConnectAuthenticationHandler.cs

```diff
@@ -235,180 +276,215 @@ protected override AuthenticationTicket AuthenticateCore()
             {
                 IFormCollection form = await Request.ReadFormAsync();
```

Comment:
This code is fine. Yes ReadFormAsync caches the results internally, but he's not really caching here, he's just making each individual line easy to read by using intermediate variables.